pkg/resources/resource_sso_domain.go

```diff
@@ -105,16 +112,36 @@ func ssoDomainRead(ctx context.Context, d *schema.ResourceData, meta interface{}
 	}
 
 	// Extract sso_config_id and domain from the Terraform resource data
-	ssoConfigID := d.Get("sso_config_id").(string)
-	domainName := d.Get("domain").(string)
+	var ssoConfigID, domainName string
+
+	// Check if the ID contains a delimiter indicating it's a combined ID from an import
+	if strings.Contains(d.Id(), ":") {
+		parts := strings.Split(d.Id(), ":")
+		if len(parts) != 2 {
+			return diag.Errorf("Invalid ID format. Expected 'ssoConfigID:domainName'")
+		}
+		ssoConfigID, domainName = parts[0], parts[1]
+		// Set the sso_config_id attribute
+		if err := d.Set("sso_config_id", ssoConfigID); err != nil {
+			return diag.FromErr(err)
+		}
+		// Set the domain attribute
+		if err := d.Set("domain", domainName); err != nil {
+			return diag.FromErr(err)
+		}
+	} else {
+		// Regular operation, extract sso_config_id and domain from the Terraform resource data
+		ssoConfigID = d.Get("sso_config_id").(string)
+		domainName = d.Get("domain").(string)
```

Comment:
If we hit this else block, what is the value of Id? Curious why it isn't relevant here?

Reply:
Yes, good question! The limitation here is that Frontegg does have a separate API endpoint for creating and deleting domains, but does not have a direct API endpoint for getting the domains, so to get the list of the domains, you need to call the SSO config endpoint.

By default, if this is a regular operation, eg, creating a new domain or just refreshing the state, the domain will be available from the state: `domainName = d.Get("domain").(string)`, but when we do the import, we don't have anything in the state so we need a way to pass it along, hence the `ssoConfigID:domainName` syntax where we would extract both the domain and the sso config ID from.

Reply:
That makes sense. Is this a standard trick in Terraform land? I've never seen it before, but also I've imported like less than a dozen resources ever in my life.

Reply:
Actually now thinking about this more, I've refactored this to use custom import functions, that way the read function is simplified for regular operations, and that logic is abstracted to the import function. Seems like I overengineered this yesterday 🤦